superset/db_engine_specs/hana.py

```diff
@@ -42,9 +42,24 @@ class HanaEngineSpec(PostgresBaseEngineSpec):
 
     @classmethod
     def convert_dttm(cls, target_type: str, dttm: datetime) -> Optional[str]:
+        """
+        Tested: hana time returns DATETIME and STRING
+        HANA:
+        :return dttm->DATETIME
+        HANA->DATE:f"TO_DATE('{dttm.date().isoformat()}', 'YYYY-MM-DD')"
+
+        :return dttm->DATETIME
+        HANA->TIMESTAMP:
+        f"TO_TIMESTAMP('{dttm.isoformat(timespec="microseconds")}',
+        'YYYY-MM-DD"T"HH24:MI:SS.ff6')"
+
+        :return dttm->STRING
+        HANA->NVARCHAR TYPE:f"TO_CHAR('{dttm.date().isoformat()}', 'YYYYMMDD')"
+        """
         tt = target_type.upper()
-        if tt == "DATE":
-            return f"TO_DATE('{dttm.date().isoformat()}', 'YYYY-MM-DD')"
-        if tt == "TIMESTAMP":
-            return f"""TO_TIMESTAMP('{dttm.isoformat(timespec="microseconds")}', 'YYYY-MM-DD"T"HH24:MI:SS.ff6')"""  # pylint: disable=line-too-long
+        if tt == "DATETIME":
+            return f"""TO_TIMESTAMP('{dttm.isoformat(timespec="microseconds")}', \
+             'YYYY-MM-DD"T"HH24:MI:SS.ff6')"""  # pylint: disable=line-too-long
+        if tt == "STRING":
```

Comment:
We don’t want to handle the generic string conversion here as it depends on the string format.

Could you update the unit tests as well (which may serve as a better mechanism for documenting this logic).

Reply:
Most of the hana data comes from SAP, and the SAP time data store is a string by default in oracle. If the data engineer does not process the time field in the hana model, then superset will not support time filtering and can only hardcode the filter conditions

Reply:
Can you explain this in more detail? I'm having a hard time following how this works, as I'm unfamiliar with the internals of SAP/Hana.

Reply:
> Can you explain this in more detail? I'm having a hard time following how this works, as I'm unfamiliar with the internals of SAP/Hana.

![image](https://user-images.githubusercontent.com/44701753/68987354-04e3a100-0864-11ea-8a11-8608475ec258.png)
![image](https://user-images.githubusercontent.com/44701753/68987357-0f059f80-0864-11ea-909e-89b0a4b794c0.png)
![image](https://user-images.githubusercontent.com/44701753/68987373-3e1c1100-0864-11ea-84d0-47a77a1fc157.png)
![image](https://user-images.githubusercontent.com/44701753/68987388-71f73680-0864-11ea-9d06-8bb745e27e21.png)
![image](https://user-images.githubusercontent.com/44701753/68987393-9226f580-0864-11ea-81e4-85e57379acaf.png)
![image](https://user-images.githubusercontent.com/44701753/68987398-ae2a9700-0864-11ea-8f7b-732a316a19bc.png)


I think the screenshot should illustrate all this

Reply:
SAP saves the time as a string, and hana reads naturally as a string, unless it is deliberately processed. If the data engineer does not deliberately process it as a time format when selecting this column, then superset cannot filter by time.


![image](https://user-images.githubusercontent.com/44701753/68987495-a91a1780-0865-11ea-9b43-4307505b135c.png)